src/base/run_state_intf.mli

```diff
@@ -0,0 +1,89 @@
+module Vector : sig
```

Comment:
Wondering if making an external library for vectors (and the other types we use in Pickles) would make sense. Could be nice for the OCaml community too.

Reply:
Not blocking, but I don't understand why we have the nested module `Vector` with the methods, and externally we do have only the type, both types being equal.